nixos/modules/services/development/zammad.nix

```
@@ -281,54 +294,51 @@ in
       };
       after = [
         "network.target"
+        "systemd-tmpfiles-setup.service"
+      ] ++ optionals (cfg.database.createLocally && cfg.database.type == "PostgreSQL") [
         "postgresql.service"
       ] ++ optionals cfg.redis.createLocally [
         "redis-${cfg.redis.name}.service"
       ];
-      requires = [
+      requires = optionals (cfg.database.createLocally && cfg.database.type == "PostgreSQL") [
         "postgresql.service"
       ];
       description = "Zammad web";
       wantedBy = [ "multi-user.target" ];
       preStart = ''
-        # Blindly copy the whole project here.
-        chmod -R +w .
-        rm -rf ./public/assets/
-        rm -rf ./tmp/*
-        rm -rf ./log/*
-        cp -r --no-preserve=owner ${cfg.package}/* .
-        chmod -R +w .
         # config file
-        cp ${databaseConfig} ./config/database.yml
-        chmod -R +w .
+        cat ${databaseConfig} > ${cfg.dataDir}/config/database.yml
         ${optionalString (cfg.database.passwordFile != null) ''
         {
           echo -n "  password: "
           cat ${cfg.database.passwordFile}
-        } >> ./config/database.yml
+        } >> ${cfg.dataDir}/config/database.yml
         ''}
         ${optionalString (cfg.secretKeyBaseFile != null) ''
         {
           echo "production: "
           echo -n "  secret_key_base: "
           cat ${cfg.secretKeyBaseFile}
-        } > ./config/secrets.yml
+        } > ${cfg.dataDir}/config/secrets.yml
         ''}
 
-        if [ `${config.services.postgresql.package}/bin/psql \
-                  --host ${cfg.database.host} \
-                  ${optionalString
-                    (cfg.database.port != null)
-                    "--port ${toString cfg.database.port}"} \
-                  --username ${cfg.database.user} \
-                  --dbname ${cfg.database.name} \
-                  --command "SELECT COUNT(*) FROM pg_class c \
-                            JOIN pg_namespace s ON s.oid = c.relnamespace \
-                            WHERE s.nspname NOT IN ('pg_catalog', 'pg_toast', 'information_schema') \
-                              AND s.nspname NOT LIKE 'pg_temp%';" | sed -n 3p` -eq 0 ]; then
+        # needed for cleanup
+        shopt -s extglob
+
+        # cleanup state directory from module before refactoring
```

"before refactoring" is not going to be clear looking at this code outside the context of the PR.
```suggestion
        # cleanup state directory from an earlier version of this module
```

But is this even necessary? Are the extra files hurting anything? If not, I would remove this to simplify the module. Otherwise, I wonder if we can phase it out after a certain amount of time.

---

I have linked this PR there, so now it should be clear which refactoring I mean. And I think we can remove the cleanup after a year or so, but until then I would like to keep the cleanup in the module.